mslib/mscolab/server.py

```diff
@@ -351,6 +356,40 @@ def get_user():
     return json.dumps({'user': {'id': g.user.id, 'username': g.user.username}})
 
 
+@APP.route('/upload_profile_image', methods=["POST"])
+@verify_user
+def upload_profile_image():
+    user_id = request.form['user_id']
+    file = request.files['image']
+    if file and allowed_file(file.filename):
+        if file.mimetype.startswith('image/'):
+            try:
+                data = file.read()
+                if len(data) > mscolab_settings.MAX_IMAGE_SIZE:
+                    return jsonify({'message': 'File too large'}), 413
+                success, message = fm.save_user_profile_image(user_id, data)
+                if success:
+                    return jsonify({'message': message}), 200
+                else:
+                    return jsonify({'message': message}), 400
+            except Exception as e:
+                return jsonify({'message': str(e)}), 500
+        else:
+            return jsonify({'message': 'Invalid file type'}), 400
+    else:
+        return jsonify({'message': 'No file provided or invalid file type'}), 400
+
+
+@APP.route('/fetch_profile_image', methods=["GET"])
+def fetch_profile_image():
+    user_id = request.form['user_id']
```

Comment:
There is not a protection for anyone using curl to fetch all images from the database.

When you use the method we defined for the uploaded chat files one has to guess a random string too.